### DISCUSSION TITLE
Implement host and two VMs' serial port output and input emulation.

### DISCUSSION THREAD

**Discussion by pengzechen** [arceos-hypervisor/axvisor#9] [General]
I want to start three cores to do that.
core 0 runs vm 0, and core 1 runs vm1
core2 runs a shell that can change serial

**Comment by hky1999** [comment]
You'd better implement this  demo under this [branch](https://github.com/arceos-hypervisor/arceos-umhv/tree/task_vcpu) with axtask.

Still, your design seems strange, ask more dalaos for suggestions. @aarkegz

**Reply by pengzechen** [reply]
emm, "/tree/task_vcpu" couldn't work now. （can't even do exception handling）
ok

**Comment by pengzechen** [comment]
@aarkegz Vm managing shell should run in el2 in host?

**Reply by hky1999** [reply]
of course, if there has to be one.
